### PR TITLE
test: pin auth edge-case behavior after Hono migration (#438)

### PR DESCRIPTION
**Pull request by takumin**
## Summary

Follow-up to the two P2 Codex review comments left on PR #438 (the merged Hono migration). Both flag auth edge cases whose behavior changed under Hono, contrary to that PR's "behavior-preserving" claim:

- **[#438 review 1](https://github.com/takumin/cloudflare-workers-github-apps-playground/pull/438#discussion_r3644202537)** — The auth middleware (`app.use("*")`) runs before route matching, so with `API_TOKEN` set an unauthenticated `GET /other` or `POST /` now returns `401` instead of the previous JSON `404`.
- **[#438 review 2](https://github.com/takumin/cloudflare-workers-github-apps-playground/pull/438#discussion_r3644202543)** — Hono's `bearerAuth` returns `400` (not the previous `401`) for a non-Bearer `Authorization` header (e.g. `Token ...`, or a bare `Bearer` with no token); the `invalidAuthenticationHeader` option only changes the body, not the status.

Both observations are valid when compared against the pre-Hono hand-rolled implementation from PR #432. The decision here is to **accept these `401`/`400` responses as the intended behavior** rather than revert them: no production code changes, only **regression tests that pin the current Hono behavior** so any future change is deliberate and detected by the suite.

## What changed

- **`test/index.spec.ts`**: added the 4 regression tests below (`src/index.ts` is untouched). The intent is documented in a comment on the added block.
  - Non-root path + `API_TOKEN` set + no Authorization header → `401 { "error": "Unauthorized" }`
  - Non-GET method + `API_TOKEN` set + no Authorization header → `401 { "error": "Unauthorized" }`
  - Non-Bearer `Authorization: Token ...` + `API_TOKEN` set → `400 { "error": "Unauthorized" }`
  - Bare `Authorization: Bearer` (no token) + `API_TOKEN` set → `400 { "error": "Unauthorized" }`

## Test plan

- [x] `vitest run` — 14 tests pass (10 existing + 4 new)
- [x] `@biomejs/biome check test/index.spec.ts` — clean
- [x] `oxlint test/ src/` — 0 warnings / 0 errors
- [x] `tsc --noEmit -p test/tsconfig.json` — clean

Note: the sandbox's Node is `v22.22.2` (the repo pins `24.18.0`) and `aqua` isn't available, so install used `pnpm install --ignore-scripts --config.engine-strict=false` and each tool was run via `npx` / the local binary. Please treat CI's normal `pnpm run test` / `pnpm run lint` as authoritative.

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: test/index.spec.ts [hunk 1]
```diff
@@ -215,4 +215,80 @@ describe("GitHub Apps API worker", () => {
...
 		expect(await response.json()).toEqual(mockRepositoriesData);
 	});
+
+	// The Hono migration (#438) enforces Bearer auth via a catch-all
+	// `app.use("*")` that runs before route matching, and delegates the header
+	// check to Hono's bearerAuth middleware. Codex flagged two behavior changes
+	// from the pre-Hono exact-match contract (#438 review): auth now runs before
+	// the 404 route guard, and a non-Bearer Authorization header yields 400
+	// instead of 401. Both were accepted as the intended behavior; the tests
+	// below pin it so any future change is deliberate.
```
[the remark this answers is not in the document]
Comment (takumin) [reply test/index.spec.ts:225]: Good catch — addressed in 7b431d1.

- `README.md`: clarified that with `API_TOKEN` set, auth runs before route matching, so an unauthenticated request returns `401` (missing/wrong token) or `400` (malformed non-`Bearer` header) regardless of path/method; the JSON `404` for non-`GET /` applies when auth is unset or already passed.
- `wrangler.toml`: noted that a malformed non-`Bearer` `Authorization` header returns `400` (from Hono's `bearerAuth`), not `401`.

The docs now match the statuses pinned by the new tests.

---
_Generated by [Claude Code](https://claude.ai/code)_

**Comment by takumin** [comment]
@codex review

The docs inconsistency flagged in the previous review has been addressed in 7b431d1 (`README.md` and `wrangler.toml` now match the auth/routing status codes pinned by the new tests).

---
_Generated by [Claude Code](https://claude.ai/code)_
